pilot/pkg/bootstrap/server.go

```diff
@@ -1130,30 +1130,6 @@ func (s *Server) startCA(caOpts *caOptions) {
 	})
 }
 
-func (s *Server) fetchCARoot() map[string]string {
-	if s.isDisableCa() {
-		return map[string]string{
-			constants.CACertNamespaceConfigMapDataName: string(s.RA.GetCAKeyCertBundle().GetRootCertPem()),
-		}
-	}
-	if s.CA == nil {
-		return nil
-	}
-
-	// For Kubernetes CA, we don't distribute it; it is mounted in all pods by Kubernetes.
-	if features.PilotCertProvider == constants.CertProviderKubernetes {
-		return nil
-	}
-	// For no CA we don't distribute it either, as there is no cert
-	if features.PilotCertProvider == constants.CertProviderNone {
-		return nil
-	}
```

Comment:
Sounds like we are losing this functionality which will break a lot of users?

Reply:
IC, in this case, ns controller should not be started